collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -105,13 +113,14 @@ protected void onDetachedFromWindow() {
         }
     }
 
-    @Override
-    public void onButtonClick(int buttonId) {
-        if (!isUrlEmpty(stringAnswer)) {
-            customTabHelper.bindCustomTabsService(getContext(), null);
-            customTabHelper.openUri(getContext(), uri);
-        } else {
-            Toast.makeText(getContext(), "No URL set", Toast.LENGTH_SHORT).show();
+    public void onButtonClick() {
+        if (MultiClickGuard.allowClick(QuestionWidget.class.getName())) {
```

Comment:
This check will be needed in case of every single widget button. If so it would be better to create our own `WidgetButton` class that will extend `com.google.android.material.button.MaterialButton` (then you can use it in your layout file). In such a class we can then override:
```
    @Override
    public boolean performClick() {
        if (MultiClickGuard.allowClick(getClass().getName())) {
            return super.performClick();
        } else {
            return false;
        }
    }

what do you think about this approach?
```

Reply:
Yes, that would be great. I can create `WidgetButton` class in the `material` package?

Reply:
Maybe later, we can shift more common functionality to the `WidgetButton` class.

Reply:
> class in the material package

i'm not sure that package contains implementations for elements that don't exist the Material Components framework in this case it's about customizing not implementing so maybe our main dir would be a better choice what do you think @seadowg

Reply:
Yeah the `material` module is mostly for implementations of Material Components like @grzesiek2010 suggests. I'd say we could make a `MulticlickSafeButton` that extends `MaterialButton` and just lives in the `org.odk.collect.android.views` seeing as it's just a general view. If it seems like like we want one just for widgets it could live in the `widgets` package.

Reply:
@seadowg @grzesiek2010, Should I go ahead and make `MulticlickSafeButton` in the `org.odk.collect.android.views` package?

Reply:
Yes please go ahead you will need it in other widgets either way so I think it's a good idea